packages/holocron/src/loadModule.node.js

```diff
@@ -37,75 +61,288 @@ function fetchModuleContent(moduleUrl) {
+
+  if (messages.length > 0) {
+    if (moduleCanBeSafelyLoaded || (process.env.ONE_DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS === 'true')) {
+      // eslint-disable-next-line no-console
+      console.warn(messages.join('\n'));
+    } else {
+      throw new Error(messages.join('\n'));
+    }
+
+    setModulesRequiredExternals({ moduleName, externals: moduleExternals });
+  }
+  registerModuleUsingExternals(moduleName);
+};
+
+/**
+ * Fetches Module Config. Returns null if does not exist or an error has occurred
+ * @param {string} baseUrl path to the assets
+ * @returns {Promise<object | null>}
+ */
+const fetchModuleConfig = async (baseUrl) => {
+  const moduleConfigUrl = `${baseUrl}module-config.json`;
 
-function checkStatus(response) {
-  if (response.status >= 200 && response.status < 300) {
-    return response;
+  try {
+    const moduleConfigStr = await fetchAsset(moduleConfigUrl);
+    const moduleConfig = JSON.parse(moduleConfigStr);
+
+    return moduleConfig;
+  } catch (err) {
+    console.warn('Module Config failed to fetch and parse, external fallbacks will be ignored.', err);
   }
 
-  const error = new Error(response.statusText);
-  error.response = response;
-  throw error;
-}
+  return null;
+};
 
-export default async function loadModule(
+const loadModule = async (
   moduleName,
-  { node: { integrity, url } },
+  { node: { integrity, url }, baseUrl },
   onModuleLoad = () => null
-) {
+) => {
+  let moduleConfig;
+  const oldRequiredExternals = getRequiredExternalsRegistry()[moduleName];
+  clearModulesRequiredExternals(moduleName);
   try {
     assert(typeof moduleName === 'string', 'moduleName must be a string');
 
     if (isModuleInBlockList(url)) {
       throw new Error(`module at ${url} previously failed to load, will not attempt to reload.`);
     }
 
-    const moduleResponse = await fetchModuleContent(url);
-    let nodeModule;
+    const rootModule = global.getTenantRootModule ? global.getTenantRootModule() : null;
 
-    try {
-      checkStatus(moduleResponse);
-      const moduleString = await moduleResponse.text();
-      if (process.env.NODE_ENV === 'production') {
-        const actualSRI = ssri.fromData(
-          moduleString,
-          { algorithms: ['sha256', 'sha384'] }
-        ).toString();
-
-        assert(integrity === actualSRI, `SRI for module at ${url} must match SRI in module map.\n Expected ${integrity}, received ${actualSRI}`);
-      }
-      nodeModule = requireFromString(moduleString, url);
-    } catch (err) {
-      if (process.env.NODE_ENV === 'development') {
-        console.warn('Holocron module "%s" at "%s" failed to execute.\n\t[Error Message]: "%s"\nPlease fix any errors and wait for it to be reloaded.', moduleName, url, err.message);
-      } else if (err.shouldBlockModuleReload !== false) {
-        addToModuleBlockList(url);
-        console.warn(`Holocron module at ${url} added to blocklist: ${err.message}`);
+    // if no root module, module being loaded should be root.
```

Comment:
Is the root module always loaded first? If not, we may miss some externals here

Reply:
yes, it has to be